lib/contentful_rails/engine.rb

```diff
@@ -23,8 +23,10 @@ class Engine < ::Rails::Engine
     #Iterate through all models which inherit from ContentfulModel::Base
     #and add an entry mapping for them, so calls to the Contentful API return
     #the appropriate classes
+    #If ContentfulModel.configuration.entry_mapping is already configured from
+    #initializer, assuming that is the one to use and not running `add_entry_mapping`
     initializer "add_entry_mappings", after: :configure_contentful  do
-      if defined?(ContentfulModel)
+      if defined?(ContentfulModel) && ContentfulModel.configuration.entry_mapping.blank?
```

Comment:
I would change this to be per descendant, so if people define a mapping for one of their models and not for every model, you can easily create the default mapping this gem provides. Else, you would be losing most of the `contentful_model` power.

Sadly I do not know how to circumvent eager loading in that case. So, What I think that might be the best of both worlds is, to add an option in the configuration to manually prevent eager loading (with a proper explanation of what they would be losing in the README) and use that instead of entry mapping existence to determine wether to eager load or not, and then check the entry mapping per descendent to use the one in the config or use the default `contentful_model` provided one.

Leaving in that case the responsibility to the user to determine wether it's worth it or not to do this. And explain them how to make it work properly when using this option.